### PR TITLE
Update wasm threads proposal tests and fix cmpxchg_u

### PR DESCRIPTION
**Pull request by justinmichaud**
#### 6b5f20366027ae0df88a2f78a8362c10df6ddfd8
<pre>
Update wasm threads proposal tests and fix cmpxchg_u
<a href="https://bugs.webkit.org/show_bug.cgi?id=263292">https://bugs.webkit.org/show_bug.cgi?id=263292</a>
rdar://117102231

Reviewed by NOBODY (OOPS!).

In <a href="https://github.com/WebAssembly/threads/issues/195">https://github.com/WebAssembly/threads/issues/195</a>, it was documented
that JSC implements cmpxchg_u without wrapping the expected value.

Previously, the spec tests and reference interpreter followed this interpretation,
but the spec text, SpiderMonkey and v8 did the wrapping.

The spec tests were updated, and this new behaviour is simpler anyway,
so let&apos;s just do it.

* JSTests/wasm/spec-tests/atomic.wast.js: Added.
* JSTests/wasm/spec-tests/binary-leb128.wast.js:
* JSTests/wasm/spec-tests/binary.wast.js:
* JSTests/wasm/spec-tests/br_table.wast.js:
* JSTests/wasm/spec-tests/elem.wast.js:
* JSTests/wasm/spec-tests/float_literals.wast.js:
* JSTests/wasm/threads-spec-tests/atomic-signed.wast.js:
* Source/JavaScriptCore/llint/WebAssembly64.asm:
* Source/JavaScriptCore/wasm/WasmB3IRGenerator.cpp:
(JSC::Wasm::B3IRGenerator::emitAtomicCompareExchange):
* Source/JavaScriptCore/wasm/WasmBBQJIT.cpp:
(JSC::Wasm::BBQJIT::emitAtomicCompareExchange):
</pre>
https://github.com/WebKit/WebKit/commit/6b5f20366027ae0df88a2f78a8362c10df6ddfd8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22585 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/240 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/23669 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/24491 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20908 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/2080 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/23116 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/21872 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22824 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/440 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19594 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/25344 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/363 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/20457 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26713 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/19666 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/20740 "Found 1 new API test failure: TestWebKitAPI.ServiceWorker.WindowClientNavigate (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20702 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/24547 "Passed tests") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/21953 "Found 5 new JSC stress test failures: wasm.yaml/wasm/spec-tests/atomic.wast.js.default-wasm, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-collect-continuously, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-eager-jettison, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-no-cjit, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-omg (failure)") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/148 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18009 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/26003 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/116 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/6101 "Found 4 new JSC stress test failures: wasm.yaml/wasm/spec-tests/atomic.wast.js.default-wasm, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-collect-continuously, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-eager-jettison, wasm.yaml/wasm/spec-tests/atomic.wast.js.wasm-no-cjit (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/168 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/27282 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/164 "Built successfully") | | [loading 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5927 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Passed JSC tests (warnings)") |

### CODE REVIEW

File: JSTests/wasm/spec-tests/elem.wast.js [hunk 1]
```diff
@@ -287,3 +287,16 @@ assert_return(() => call($m, "get", [0]), null);
...
 assert_return(() => call($m, "get", [1]), externref(137));
+
+// elem.wast:681
+let $30 = instance("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x85\x80\x80\x80\x00\x01\x60\x00\x01\x7f\x03\x82\x80\x80\x80\x00\x01\x00\x06\x86\x80\x80\x80\x00\x01\x70\x00\xd2\x00\x0b\x07\x85\x80\x80\x80\x00\x01\x01\x66\x03\x00\x0a\x8a\x80\x80\x80\x00\x01\x84\x80\x80\x80\x00\x00\x41\x2a\x0b");
+let $module4 = $30;
+
+// elem.wast:688
+register("module4", $module4)
+
+// elem.wast:690
```
Comment (justinmichaud) [inline JSTests/wasm/spec-tests/elem.wast.js:298]: We need to figure out why we fail this

File: JSTests/wasm/threads-spec-tests/atomic-signed.wast.js [hunk 2]
```diff
@@ -671,7 +671,7 @@ run(() => call(instance("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x89\x80\x80\x80\x0
 run(() => call(instance("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x88\x80\x80\x80\x00\x02\x60\x00\x00\x60\x01\x7e\x00\x02\x8b\x80\x80\x80\x00\x01\x02\x24\x31\x04\x69\x6e\x69\x74\x00\x01\x03\x82\x80\x80\x80\x00\x01\x00\x07\x87\x80\x80\x80\x00\x01\x03\x72\x75\x6e\x00\x01\x0a\x99\x80\x80\x80\x00\x01\x93\x80\x80\x80\x00\x00\x02\x40\x42\xf0\xe1\xc3\x87\x8f\x9e\xbc\xf8\x70\x10\x00\x0f\x0b\x00\x0b", exports("$1", $1)),  "run", []));  // run(() => call($1, "init", [int64("-1_085_102_592_571_150_096")]))
 
 // atomic-signed.wast:402
-assert_return(() => call($1, "i32.atomic.rmw8.cmpxchg_u", [0, -252_645_136, 17]), 240);
+// assert_return(() => call($1, "i32.atomic.rmw8.cmpxchg_u", [0, -252_645_136, 17]), 240);
```
Comment (justinmichaud) [inline JSTests/wasm/threads-spec-tests/atomic-signed.wast.js:674]: This test is no longer included upstream, so I just commented out the bad tests

**Review by Constellation — APPROVED** [review]
r=me

**Review by kmiller68 — CHANGES_REQUESTED** [review]
Can we update the IPInt while we're at it? Note, there's no testing in automation for IPInt so you'll have to run the spec tests by hand with `--useWasmIPInt=1`

**Pull request closed without merging**